src/cls/rbd/cls_rbd.cc

```diff
@@ -5052,8 +5052,8 @@ int trash_list(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
   CLS_LOG(20, "trash_get_images");
   while (data.size() < max_return) {
     map<string, bufferlist> raw_data;
-    int max_read = std::min<int32_t>(RBD_MAX_KEYS_READ,
-                                     max_return - data.size());
+    max_read = std::min<int32_t>(RBD_MAX_KEYS_READ,
+                                 max_return - data.size());
     int r = cls_cxx_map_get_vals(hctx, last_read, trash::IMAGE_KEY_PREFIX,
```

Comment:
@joscollin ,  reference to PR [#16667:5049](https://github.com/ceph/ceph/pull/16667/files#diff-e393a69c5ca0d5ea740c0d60d68e55f5R5049) , since `max_return` could be specified from client, `max_read` will be reassigned at L5055. so... fix this by dropping L5049 seems to be better, what do you think?